### ISSUE TITLE
crash at showing progress dialog

### ISSUE THREAD

**Issue by AhmadullahSaikat** [Kaopiz/KProgressHUD#48]
android.view.WindowLeaked: Activity  has leaked window com.android.internal.policy.impl.PhoneWindow$DecorView{360ddfca V.ED.... R......D 0,0-108,108} that was originally added here

**Comment by michzio** [comment]
Yes I have the same issue it happens when there is Screen rotation. I have spinning progress then rotate, and in the same time there should be success KProgressHUD ... .show()

**Comment by michzio** [comment]
Maybe here: 

public KProgressHUD show() {
        if (!isShowing()) {
            mFinished = false;
            if (mGraceTimeMs == 0) {
                mProgressDialog.show();
            } else {
                mGraceTimer = new Handler();
                mGraceTimer.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        if (mProgressDialog != null && !mFinished) {
                            mProgressDialog.show();
                        }
                    }
                }, mGraceTimeMs);
            }
        }
        return this;
    }

Should be something like this 

 val activity = (context as? Activity)
            if(activity != null && activity.window.decorView.isShown) {
                //Show Your KProgressHUD

}

I tested outside as wrapper of hud.show() and it sometimes helps but not always as sometimes activity.window.decorView.isShown return true and as before detaching.

**Comment by zhangchaoxu** [comment]
I encountered the same issue
